tests/unit/lib/deploy/test_deployer.py

```diff
@@ -342,6 +348,13 @@ def test_wait_for_changeset(self):
         self.deployer._client.get_waiter = MagicMock(return_value=MockChangesetWaiter())
         self.deployer.wait_for_changeset("test-id", "test-stack")
 
+    @patch("os.environ", {**os.environ, "SAM_CLI_POLL_DELAY": 10})
+    def test_wait_for_changeset_client_sleep(self):
+        deployer = Deployer(MagicMock().client("cloudformation"), client_sleep=os.getenv("SAM_CLI_POLL_DELAY", 0.5))
+        deployer._client.get_waiter = MagicMock(return_value=MockChangesetWaiter())
+        deployer.wait_for_changeset("test-id", "test-stack")
+        self.assertEqual(deployer.client_sleep, 10)
+
```

Comment:
Can we also add a test to test that the `Delay` value in the waiter config is set correctly? This can be done by checking if the waiter's `.wait` call was called with the correct parameters.

Reply:
Thanks @lucashuy for the review. Yeah definitely makes sense. I will push that change.